storage/host-flash/host-flash.c

```diff
@@ -51,28 +52,37 @@ ssize_t hostflash_read(unsigned int addr, void *buff, size_t bufflen)
 
 ssize_t hostflash_write(unsigned int addr, void *buff, size_t bufflen)
 {
-    char currentByte = 0;
-    size_t wrote = 0;
+    char tempTab[256];
+    size_t wrote = 0, i;
+    int chunkLen;
     ssize_t stat;
 
     if (addr + bufflen > hostflash_common.flashsz)
         return -EINVAL;
 
     while (wrote != bufflen) {
-        stat = hostflash_read(addr + wrote, &currentByte, 1);
+        if ((bufflen - wrote) / sizeof(tempTab))
+            chunkLen = sizeof(tempTab);
+        else
+            chunkLen = (bufflen - wrote) % sizeof(tempTab);
+
+        stat = hostflash_read(addr + wrote, tempTab, chunkLen);
         if (stat <= 0)
             return stat;
 
-        currentByte = currentByte & *((char *)buff + wrote);
-        stat = pwrite(hostflash_common.filefd, &currentByte, 1, addr + wrote);
+        for (i = 0; i < stat; i++) {
+            tempTab[i] = tempTab[i] & *((char *)buff + wrote + i);
+        }
```

Comment:
According to our coding convention:
_Opening and closing braces should be used only if the body of the conditional instruction is longer than one line_

Reply:
Done